pkgs/development/python-modules/snappy/default.nix

```
@@ -0,0 +1,119 @@
+{
+  lib,
+  fetchpatch,
+  python,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  cypari,
+  cython,
+  fxrays,
+  ipython,
+  libGL,
+  libX11,
+  low-index,
+  packaging,
+  pickleshare,
+  plink,
+  pypng,
+  pyx,
+  snappy-15-knots,
+  snappy-manifolds,
+  spherogram,
+  sphinx,
+  sphinx-rtd-theme,
+  tkinter-gl,
+}:
+
+buildPythonPackage rec {
+  pname = "snappy";
+  version = "3.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "3-manifolds";
+    repo = "SnapPy";
+    tag = "${version}_as_released";
+    hash = "sha256-IwPxuyVDsL5yML+J06HTKlz52sYrPkohLJ0XDXDwTlo=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "no-copy-doc.patch";
+      url = "https://github.com/3-manifolds/SnapPy/commit/c6aeeaaec7010a54e4ebdf2e8dad7b384c2ce8e5.patch";
+      hash = "sha256-OV3Qr5wO5UHNzVFTPTujIpp5ptel6gvAlcMgrJ8C0KY=";
+    })
+    (fetchpatch {
+      name = "fix-test-aarch64.patch";
+      url = "https://github.com/3-manifolds/SnapPy/commit/a8d57db39bc8f486746dc61027779168d0bc316a.patch";
+      hash = "sha256-RsuwaR+7UrVTKlPwQeHblTAN++La7b9BSMdFcJSiX5Q=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail "/usr/include/GL" "${libGL.dev}/include/GL" \
+      --replace-fail "'python/doc'" "'$out/${python.sitePackages}/snappy/doc'"
```

```suggestion

    substituteInPlace python/app_menus.py \
      --replace-fail "os.path.join(os.path.dirname(snappy_dir), 'doc')" "os.path.join('$doc', 'share', 'doc', '$name', 'html')"
      
```
Re: `sphinxHook`. For the help pages to be found within SnapPy.